src/Common/OptimizedRegularExpression.cpp

```diff
@@ -14,13 +14,42 @@ namespace DB
     }
 }
 
+namespace
+{
 
-template <bool thread_safe>
-void OptimizedRegularExpressionImpl<thread_safe>::analyze(
+struct Literal
+{
+    std::string literal;
+    bool prefix; /// this literal string is the prefix of the whole string.
+    bool suffix; /// this literal string is the suffic of the whole string.
+    void clear()
+    {
+        literal.clear();
+        prefix = false;
+        suffix = false;
+    }
+};
+
+using Literals = std::vector<Literal>;
+
+size_t shortest_alter_length(const Literals & literals)
```

Comment:
The Literals argument is not restricted to being alternatives. Suggest to rename this function `shortest_literal_length()`.